src/pages/index.js

```diff
@@ -4,16 +4,72 @@ import Section from "../components/Section.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import UserInfo from "../components/UserInfo.js";
-import { initialCards, validationSettings } from "../utils/constants.js";
+import Api from "../components/api.js";
+import { validationSettings } from "../utils/constants.js";
 import "../pages/index.css";
 
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "f7c32d64-03fb-4be9-a84c-fb3ea30dd569",
+    "Content-Type": "application/json",
+  },
+});
+
 // Selectors
 const cardListElement = ".cards__list";
 const profileEditModal = "#profile-edit-modal";
 const addCardModal = "#add-card-modal";
 const cardPreviewModal = ".modal_type_preview";
 const profileForm = document.querySelector("#profile-form");
 const addCardForm = document.querySelector("#card-form");
+const avatarForm = document.querySelector("#avatar-form");
+const editAvatarButton = document.querySelector("#edit-avatar-button");
+const profileImage = document.querySelector(".profile__image");
+
+// Delete confirmation modal setup
+const confirmModal = document.querySelector(".modal_type_confirm");
+const confirmForm = confirmModal.querySelector(".modal__form");
+const cancelButton = confirmModal.querySelector(".modal__cancel");
+let cardToDelete = null;
+
+// Modal handlers
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", closeOnEsc);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", closeOnEsc);
+}
+
+function closeOnEsc(evt) {
+  if (evt.key === "Escape") {
+    const openedModal = document.querySelector(".modal_opened");
+    closeModal(openedModal);
+  }
+}
+
+cancelButton.addEventListener("click", () => {
+  cardToDelete = null;
+  closeModal(confirmModal);
+});
+
+confirmForm.addEventListener("submit", (e) => {
+  e.preventDefault();
+  if (cardToDelete) {
+    const cardId = cardToDelete.getAttribute("data-id");
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        cardToDelete.remove();
+        cardToDelete = null;
+        closeModal(confirmModal);
+      })
+      .catch(console.error);
```

Comment:
The `confirmation` modal should be created from the `PopupWithConfirmation` class, according to the checklist https://snipboard.io/9BkQtq.jpg

And the `submit` handler should be passed into the constructor of the  `PopupWithConfirmation` class, like you do for the `PopupWithForm` instances